company/okrs/2020_q1.md

```diff
@@ -0,0 +1,114 @@
+   1. Distribution KR: Surface information to site admins and managers to encourage expansion/upsell
+
+1. **Code intelligence: LSIF is adopted by customers**
+   1. Code intelligence KR: At least three customers are using LSIF on their primary repositories
+   1. Code intelligence KR: At least one customer is using LSIF across all of their code in languages for which there exists an LSIF indexer
+
+1. **Code intelligence: Increase user confidence in code intelligence provided by LSIF**
+   1. Code intelligence KR: Code intelligence UI (tooltips, references panel, etc.) indicates whether or not results are precise
+   1. Code intelligence KR: Fall back to fuzzy if the file has been modified since the last LSIF upload
+   1. Code intelligence KR: Make use of LSIF data when the current line is not part of the diff since the last LSIF upload
+
+1. **Code intelligence: Make search better using precise code intel data**
+   1. Code intelligence KR: Implement search within dependencies and/or dependents of current repo
+   1. Code intelligence KR: Implement searching for defs/refs by symbol name
+
+1. **Web: Make users aware of advanced search features**
+   1. Web KR: Start tracking time to first search and search filters per query in pings
+   1. Web KR: Shorten time to first search
+   1. Web KR: Increase average number of search filters per query
+
+1. **Web: Automation is adopted by customers**
+   1. Web KR: At least four customers are using campaigns in production
+   1. Web KR: All campaign types have been used by customers in production
+
+1. **Core Services: Build features that accelerate customers through pipeline**
+   1. Core Services KR: Shorten the time from "intent to try Sourcegraph" to "whole org is using Sourcegraph" (onboarding experience)
+   1. Core Services KR: Ensure new features (Automation, ACLs, etc) are accessible and easily tryable by customers once they go GA
+
+1. **Core Services: Search has feature parity with OpenGrok**
+   1. Core Services KR: Implement AND query support
+   1. Core Services KR: Implement OR query support
+   1. Core Services KR: Implement NOT query support
+
+1. **Core Services: Existing customers are cared for**
+   1. Core Services KR: Each iteration work plan includes prioritised customer issues.
+   1. Core Services KR: Each team mate works on relevant customer issues every iteration.
+
+1. **Core Services: Automation is polished and adopted by customers**
+   1. Core Services KR: Write blog post about automation.
+   1. Core Services KR: At least four customers are running campaigns.
+   1. Core Services KR: Implement stacking diffs.
+
+1. **Core Services: Structural search is polished and adopted by customers**
+   1. Core Services KR: Write blog post about structural search.
+   1. Core Services KR: At least four customers are running structural searches.
+   1. Core Services KR: Structural search can be ran at scale (~30k repositories).
```

Comment:
@tsenart make this measurable (choose X)

```suggestion
   1. Core Services KR: Structural searches over ~30k repositories complete in less than X seconds.
```

Reply:
An explicit perf goal like this is not something that I've had in mind immediately. I think it's a good goal. Just, as I'm currently the only person working on this, my question for this is "who is going to, or would like to, commit time to this effort this quarter?" I think that some discussion needs to cover this if it'll be a team level OKR.

Reply:
To clarify: If the idea is that I have significant individual ownership over the bullets in this section, that's fine too, but then I'd rather condense it (because it feels slightly disproportionate in size to the KRs above), and /or propose other things I have in mind rather than `30K repos in X seconds`.

Reply:
@rvantonder: This perf KR is not set in stone! It was a point for discussion. These are team OKRs though, not personal. It's OK if there's a main driver behind each KR, but I'd like us to think about whether what we're proposing here is achievable as a team, not as individuals.

Considering our other OKRs, do you feel that this would be too much (even with contributions from other team mates)?

Reply:
Since there's a planned rework putting things under a more general 'search' objective, let's go from there. I think we can just drop this KR in the reworked section.